cmd/lncli/main.go

```diff
@@ -88,3 +139,17 @@ func main() {
 		fatal(err)
 	}
 }
+
+// cleanAndExpandPath expands environment variables and leading ~ in the
+// passed path, cleans the result, and returns it.
+func cleanAndExpandPath(path string) string {
+	// Expand initial ~ to OS specific home directory.
+	if strings.HasPrefix(path, "~") {
+		homeDir := filepath.Dir(lndHomeDir)
+		path = strings.Replace(path, "~", homeDir, 1)
```

Comment:
This is incorrect, see https://golang.org/pkg/strings/#Replace

The `1` parameter means that only the first instance of `~` gets replaced.